sample/ingress-controller-tls/vault/README.md

```diff
@@ -0,0 +1,184 @@
+# Using Secrets Store CSI to Enable NGINX Ingress Controller with TLS
+
+This guide demonstrates steps required to setup Secrets Store CSI driver to enable applications to work with NGINX Ingress Controller with TLS stored in an external Secrets store. 
+For more information on securing an Ingress with TLS, refer to: https://kubernetes.io/docs/concepts/services-networking/ingress/#tls
+
+# Generate a TLS Cert
+
+```bash
+openssl req -x509 -nodes -days 365 -newkey rsa:2048 \
+    -out ingress-tls.crt \
+    -keyout ingress-tls.key \
+    -subj "/CN=demo.test.com/O=ingress-tls"
+```
+
+# Store Cert in External Secrets Store Service
+- [Azure Key Vault](https://docs.microsoft.com/en-us/azure/key-vault/certificates/certificate-scenarios#import-a-certificate) 
+- [HashiCorp Vault](https://www.vaultproject.io/docs/commands#reading-and-writing-data)
```

Comment:
can we update this readme to only Vault specific things? then link to this readme from the one in ingress-controller-tls folder.